File: backend/src/main/java/com/transaction/system/model/Transaction.java
package com.transaction.system.model;

import com.transaction.system.enums.TransactionStatus;

import java.math.BigDecimal;
import java.time.LocalDate;

public record Transaction(
        LocalDate date,
        String accountNumber,
        String accountHolderName,
        BigDecimal amount,
        TransactionStatus status
) {
}

File: frontend/src/types/transaction.ts
import type { transactionSchema } from "@/schemas/transaction";
import type z from "zod";

export type Transaction = {
  date: string;
  accountNumber: string;
  accountHolderName: string;
  amount: number;
  status: "Pending" | "Settled" | "Failed";
};

export type CreateTransactionRequest = Omit<Transaction, "status">;

export type SortKey =
  | "default"
  | "date"
  | "accountNumber"
  | "accountHolderName"
  | "amount"
  | "status";

export type SortConfig = {
  key: SortKey;
  direction: "asc" | "desc";
};

export type AddTransactionResult = {
  success: boolean;
  message: string;
};

export type TransactionFormData = z.infer<typeof transactionSchema>;
